enforcer/src/main/java/org/wso2/micro/gateway/enforcer/security/jwt/issuer/HttpTokenServerHandler.java

```diff
@@ -0,0 +1,137 @@
+import java.util.Base64;
+
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaderNames.CONTENT_LENGTH;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaderValues.CLOSE;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaderValues.KEEP_ALIVE;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaderValues.TEXT_PLAIN;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaders.Names.CONNECTION;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpHeaders.Names.CONTENT_TYPE;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.grpc.netty.shaded.io.netty.handler.codec.http.HttpResponseStatus.UNAUTHORIZED;
+
+/**
+ * This is the http token server handler implementation.
+ */
+public class HttpTokenServerHandler extends SimpleChannelInboundHandler<HttpObject> {
+    private static TokenIssuer tokenIssuer;
+    private static boolean isAuthorized = false;
+    private static final Logger logger = LogManager.getLogger(HttpTokenServerHandler.class);
+
+    @Override
+    public void channelReadComplete(ChannelHandlerContext ctx) {
+        ctx.flush();
+    }
+
+    @Override
+    public void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Exception {
+        FullHttpResponse response = null;
+
+        if (msg instanceof HttpRequest) {
+            HttpRequest req = (HttpRequest) msg;
+            boolean keepAlive = HttpUtil.isKeepAlive(req);
+
+            String authHeader = req.headers().get("Authorization");
+
+            if (authHeader == null) {
+                isAuthorized = false;
+                response = new DefaultFullHttpResponse(req.protocolVersion(), UNAUTHORIZED);
+                response.headers().set(CONTENT_TYPE, TEXT_PLAIN);
+                logger.error(APIConstants.StatusCodes.UNAUTHORIZED.getCode() +
+                        " User is NOT authorized to generate a token. " +
+                                "Please provide a valid Authorization header to continue.");
+            } else if (authHeader.toLowerCase().startsWith("basic")) {
+                    // Authorization: Basic base64credentials
+                    String base64Credentials = authHeader.substring("Basic".length()).trim();
+                    byte[] credDecoded = Base64.getDecoder().decode(base64Credentials);
+                    String credentials = new String(credDecoded, StandardCharsets.UTF_8);
+                    // credentials = username:password
+                    final String[] values = credentials.split(":", 2);
+
+                    CredentialDto[] predefinedCredentials = ConfigHolder.getInstance().getConfig()
+                            .getJwtUsersCredentials();
+                    for (CredentialDto cred: predefinedCredentials) {
+                        if (values[0].equals(cred.getUsername()) && values[1].equals(new String(cred.getPwd()))) {
```

Comment:
Possibility of array index out of bound exception

Reply:
This is handled in the exceptionCaught method. Do we need to handle this here again?

Reply:
Handled this.